frame/composable-traits/src/dex.rs

```diff
@@ -134,6 +134,31 @@ pub trait Amm {
 	) -> Result<Self::Balance, DispatchError>;
 }
 
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq, Eq, Copy, RuntimeDebug)]
+pub enum RewardPoolType {
+	LP,
+	PBLO,
+}
+
+impl Default for RewardPoolType {
+	fn default() -> Self {
+		RewardPoolType::PBLO
+	}
+}
+
+#[derive(
+	Encode, Decode, MaxEncodedLen, TypeInfo, Clone, Default, PartialEq, Eq, Copy, RuntimeDebug,
+)]
+pub struct StakingRewardPool<RewardPoolId> {
+	pub pool_id: RewardPoolId,
+	pub pool_type: RewardPoolType,
+}
+
+pub const MAX_REWARDS: u128 = 100_000_000_000_000_000_000_000_u128;
```

Comment:
Where does this value come from?

Reply:
just a default value, I am open for any thing more sensible.

Reply:
I think @vimukthi-git may have more insight here

Reply:
Thinking about this more, I think we need a way to not have a max reward for a pool